packages/builder/src/stores/builder/components.js

```diff
@@ -653,6 +653,18 @@ export class ComponentStore extends BudiStore {
     this.update(state => {
       state.selectedScreenId = targetScreenId
       state.selectedComponentId = newComponentId
+
+      const targetScreen = get(screenStore).screens.find(
+        screen => screen.id === targetScreenId
+      )
+
+      const componentPathIds = findComponentPath(
+        targetScreen?.props,
+        newComponentId
+      ).map(component => component._id)
+
+      componentTreeNodesStore.expandNodes(componentPathIds)
```

Comment:
Could we move all this logic into the component tree nodes store to avoid polluting the component store with it? We could have a nice function like `revealNode` or something that takes in a component ID and does this. We don't need to worry finding the right screen, because it will always be the selected screen.
I also don't think this needs to actually be inside the store update function.

Reply:
Implemented this, dunno why I didn't do this in the first place. 👍